### PR TITLE
Powershell: Add powershell_command_builder

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3165 .

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/exploiters/powershell/src/powershell_command_builder.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    servers: Sequence[str],
+    current_depth: int,
+    operating_system: OperatingSystem,
+    remote_agent_binary_destination_path: PurePath,
+    otp_provider: IAgentOTPProvider,
+):
+    if operating_system != OperatingSystem.WINDOWS:
+        raise Exception(f"Unsupported operating system: {operating_system}")
+
+    otp = SET_OTP_WINDOWS % {
```
Comment (shreyamalviya) [inline monkey/agent_plugins/exploiters/powershell/src/powershell_command_builder.py:23]:
```suggestion
    set_agent_otp_command = SET_OTP_WINDOWS % {
```

**Review by shreyamalviya — APPROVED** [review]
Approved pending one code quality change

File: monkey/agent_plugins/exploiters/powershell/src/powershell_command_builder.py (hunk 1 above)
Comment (ilija-lazoroski) [reply monkey/agent_plugins/exploiters/powershell/src/powershell_command_builder.py]: Addressed!

File: monkey/Pipfile
Comment (shreyamalviya) [inline monkey/Pipfile:1]: Why do we need this?

**Pull request merged**